crates/sui-graphql-rpc/src/types/checkpoint.rs

```diff
@@ -161,14 +163,15 @@ impl Checkpoint {
                 ..Default::default()
             })
         else {
-            return Ok(Connection::new(false, false));
+            return Ok(TransactionBlockConnection::new(false, false));
         };
 
         TransactionBlock::paginate(
-            ctx.data_unchecked(),
+            ctx,
             page,
             filter,
             self.checkpoint_viewed_at,
+            Some(100_000_000),
```

Comment:
I know technically it seems impossible that we would ever have more than 100,000,000 transactions in a single checkpoint, but I think for the sake of uniformity, we should add a scan limit parameter here, just like everywhere else.

We do not want to resort to defaults anywhere, because that will cause confusion.

Reply:
Yes agreed, just setting a default here for now to run e2e tests, since removing the defaults would incur corresponding changes on the tests